### ISSUE TITLE
[BUG] Docs Endpoint broken

### ISSUE THREAD

**Issue by davidsyoung** [theroyallab/tabbyAPI#64]
**Disclaimer:** Github Issues are **only** for code related bugs. If you do not understand how to startup or use TabbyAPI, please ask in the [Discord Server](https://discord.com/sYQxnuD7Fj)

**Describe the bug**

Running the latest TabbyAPI from main branch leads to a "Failed to load API definition" on the /docs endpoint.

**To Reproduce**
Visit http://ip:port/docs, and you'll be met with this image.

![CleanShot 2024-02-11 at 18 06 10@2x](https://github.com/theroyallab/tabbyAPI/assets/1213472/f668b338-22e5-4764-9006-ec240fb1fe93)

**Expected behavior**
Docs to load without issues.

**Logs**
```
INFO:     192.168.1.76:57562 - "GET /openapi.json HTTP/1.1" 500 Internal Server Error
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/uvicorn/protocols/http/h11_impl.py", line 412, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
  File "/usr/local/lib/python3.10/dist-packages/uvicorn/middleware/proxy_headers.py", line 84, in __call__
    return await self.app(scope, receive, send)
  File "/usr/local/lib/python3.10/dist-packages/fastapi/applications.py", line 1054, in __call__
    await super().__call__(scope, receive, send)
  File "/usr/local/lib/python3.10/dist-packages/starlette/applications.py", line 123, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/errors.py", line 186, in __call__
    raise exc
  File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/errors.py", line 164, in __call__
    await self.app(scope, receive, _send)
  File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/cors.py", line 83, in __call__
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "/usr/local/lib/python3.10/dist-packages/starlette/_exception_handler.py", line 64, in wrapped_app
    raise exc
  File "/usr/local/lib/python3.10/dist-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    await app(scope, receive, sender)
  File "/usr/local/lib/python3.10/dist-packages/starlette/routing.py", line 758, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/usr/local/lib/python3.10/dist-packages/starlette/routing.py", line 778, in app
    await route.handle(scope, receive, send)
  File "/usr/local/lib/python3.10/dist-packages/starlette/routing.py", line 299, in handle
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.10/dist-packages/starlette/routing.py", line 79, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  File "/usr/local/lib/python3.10/dist-packages/starlette/_exception_handler.py", line 64, in wrapped_app
    raise exc
  File "/usr/local/lib/python3.10/dist-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    await app(scope, receive, sender)
  File "/usr/local/lib/python3.10/dist-packages/starlette/routing.py", line 74, in app
    response = await func(request)
  File "/usr/local/lib/python3.10/dist-packages/fastapi/applications.py", line 1009, in openapi
    return JSONResponse(self.openapi())
  File "/usr/local/lib/python3.10/dist-packages/fastapi/applications.py", line 981, in openapi
    self.openapi_schema = get_openapi(
  File "/usr/local/lib/python3.10/dist-packages/fastapi/openapi/utils.py", line 475, in get_openapi
    field_mapping, definitions = get_definitions(
  File "/usr/local/lib/python3.10/dist-packages/fastapi/_compat.py", line 227, in get_definitions
    field_mapping, definitions = schema_generator.generate_definitions(
  File "/usr/local/lib/python3.10/dist-packages/pydantic/json_schema.py", line 372, in generate_definitions
    self.generate_inner(schema)
  File "/usr/local/lib/python3.10/dist-packages/pydantic/json_schema.py", line 547, in generate_inner
    json_schema = current_handler(schema)
  File "/usr/local/lib/python3.10/dist-packages/pydantic/_internal/_schema_generation_shared.py", line 36, in __call__
    return self.handler(__core_schema)
  File "/usr/local/lib/python3.10/dist-packages/pydantic/json_schema.py", line 521, in new_handler_func
    json_schema = js_modify_function(schema_or_field, current_handler)
  File "/usr/local/lib/python3.10/dist-packages/pydantic/main.py", line 609, in __get_pydantic_json_schema__
    return __handler(__core_schema)
  File "/usr/local/lib/python3.10/dist-packages/pydantic/_internal/_schema_generation_shared.py", line 36, in __call__
    return self.handler(__core_schema)
  File "/usr/local/lib/python3.10/dist-packages/pydantic/json_schema.py", line 521, in new_handler_func
    json_schema = js_modify_function(schema_or_field, current_handler)
  File "/usr/local/lib/python3.10/dist-packages/pydantic/_internal/_generate_schema.py", line 222, in modify_model_json_schema
    json_schema = handler(schema_or_field)
  File "/usr/local/lib/python3.10/dist-packages/pydantic/_internal/_schema_generation_shared.py", line 36, in __call__
    return self.handler(__core_schema)
  File "/usr/local/lib/python3.10/dist-packages/pydantic/json_schema.py", line 504, in handler_func
    json_schema = generate_for_schema_type(schema_or_field)
  File "/usr/local/lib/python3.10/dist-packages/pydantic/json_schema.py", line 1344, in model_schema
    json_schema = self.generate_inner(schema['schema'])
  File "/usr/local/lib/python3.10/dist-packages/pydantic/json_schema.py", line 547, in generate_inner
    json_schema = current_handler(schema)
  File "/usr/local/lib/python3.10/dist-packages/pydantic/_internal/_schema_generation_shared.py", line 36, in __call__
    return self.handler(__core_schema)
  File "/usr/local/lib/python3.10/dist-packages/pydantic/json_schema.py", line 504, in handler_func
    json_schema = generate_for_schema_type(schema_or_field)
  File "/usr/local/lib/python3.10/dist-packages/pydantic/json_schema.py", line 1436, in model_fields_schema
    json_schema = self._named_required_fields_schema(named_required_fields)
  File "/usr/local/lib/python3.10/dist-packages/pydantic/json_schema.py", line 1247, in _named_required_fields_schema
    field_json_schema = self.generate_inner(field).copy()
  File "/usr/local/lib/python3.10/dist-packages/pydantic/json_schema.py", line 547, in generate_inner
    json_schema = current_handler(schema)
  File "/usr/local/lib/python3.10/dist-packages/pydantic/_internal/_schema_generation_shared.py", line 36, in __call__
    return self.handler(__core_schema)
  File "/usr/local/lib/python3.10/dist-packages/pydantic/json_schema.py", line 539, in new_handler_func
    json_schema = js_modify_function(schema_or_field, current_handler)
  File "/usr/local/lib/python3.10/dist-packages/pydantic/_internal/_generate_schema.py", line 2106, in json_schema_update_func
    add_json_schema_extra(json_schema, json_schema_extra)
  File "/usr/local/lib/python3.10/dist-packages/pydantic/_internal/_generate_schema.py", line 2116, in add_json_schema_extra
    json_schema.update(to_jsonable_python(json_schema_extra))
pydantic_core._pydantic_core.PydanticSerializationError: Unable to serialize unknown type: <class 'function'>
```

**System info** (Bugs without this information will go lower on our priority list!)
 - OS: Unix
 - Python version: 3.10
 - CUDA/ROCm version: Repo Docker image

**Additional context**
I tried to troubleshoot and see exactly at which commit it happened, but was unable to see.

**Comment by bdashore3** [comment]
Fixed in a79c42f. This was due to a previous commit which tried to hijack Pydantic fields for inserting validator functions. However,  doing so also caused issues with JSON serialization for autodocs

**Comment by davidsyoung** [comment]
Ah. Got you! All fixed now this end
